### PR TITLE
Adds TemplateValidator

### PR DESCRIPTION
**Pull request by glevitzky**

### CODE REVIEW

File: extensions/amp-a4a/0.1/template-validator.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+import {AmpAdTemplateHelper} from '../../amp-a4a/0.1/amp-ad-template-helper';
+import {Services} from '../../../src/services';
+import {Validator, ValidatorResult} from './amp-ad-type-defs';
+import {getAmpAdMetadata} from './amp-ad-utils';
+import {pushIfNotExist} from '../../../src/utils/array';
+import {tryParseJson} from '../../../src/json';
+import {utf8Decode} from '../../../src/utils/bytes';
+
+/** @const {string} */
+export const AMP_TEMPLATED_CREATIVE_HEADER_NAME = 'AMP-template-amp-creative';
```
Comment (lannka) [inline extensions/amp-a4a/0.1/template-validator.js]: As stated in the design doc, we'd like to rename this to `AMP-Ad-Template-Extension`. Meantime, please make sure the we don't break AdZerk. So we might end of having both, until AdZerk getting migrated over.

File: extensions/amp-a4a/0.1/template-validator.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    // response as the creative, and downstream renderers may attempt to render
+    // it as a non-AMP creative within a cross-domain iframe.
+    if (!headers ||
+        headers.get(AMP_TEMPLATED_CREATIVE_HEADER_NAME) !== 'amp-mustache') {
+      return Promise.resolve(
+          /** @type {!./amp-ad-type-defs.ValidatorOutput} */ ({
+            creativeData: {
+              creative: body,
+            },
+            adResponseType: 'template',
```
Comment (lannka) [inline extensions/amp-a4a/0.1/template-validator.js]: is this an enum?

**Comment by glevitzky** [comment]
What's with the presubmit error: "Please review viewport service for helper methods or mark with `/*OK*/` or `/*REVIEW*/` and consult the AMP team"? I _do_ have the method marked with `/*OK*/ `.

File: extensions/amp-a4a/0.1/template-validator.js [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+} from './amp-ad-type-defs';
+import {AmpAdTemplateHelper} from '../../amp-a4a/0.1/amp-ad-template-helper';
+import {Services} from '../../../src/services';
+import {getAmpAdMetadata} from './amp-ad-utils';
+import {pushIfNotExist} from '../../../src/utils/array';
+import {tryParseJson} from '../../../src/json';
+import {utf8Decode} from '../../../src/utils/bytes';
+
+/** @const {string} */
+export const AMP_TEMPLATED_CREATIVE_HEADER_NAME = 'AMP-Ad-Template-Extension';
```
Comment (lannka) [inline extensions/amp-a4a/0.1/template-validator.js]: As I mentioned, can we accept both headers for now? So later we can easily migrate AdZerk over.

Comment (glevitzky) [reply extensions/amp-a4a/0.1/template-validator.js]: Done. Sorry about the confusion.

**Comment by glevitzky** [comment]
Addresses #15464.

**Pull request merged**